numba/types/scipytypes.py

```diff
@@ -0,0 +1,27 @@
+"""
+Typing for scipy specific features
+"""
+from numba import types
+
```

Comment:
This creates an import cycle, which appears to be handled properly in Python 3, but not Python 2.